### DISCUSSION TITLE
Add 4k and 8k option for both the stereo processing width field and the output size limit field

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#180] [General]
Hi, can you please add an official 4k and 8k option for both the stereo processing width field and the output size limit field under the drop downs. This would be an amazing option for those who use short 8k videos and those who use 4k sources to convert from that are doing 4k 3d and 8k 3d. Thanks

**Comment by nagadomi** [comment]
The reason I added `Stereo Processing Width` is because `row_flow_v3` does not support depth width>1920. So specifying a value higher than 1920 is a wrong choice.

For `Output Size Limit`, it is an option to work around the limitations of the cardboard's mobile hardware decoder (it cannot play videos larger than 1920x1080).
If you want to resize the source frame size, it is better to use `vf` option.
example
`scale=-1:2160` : resize to 4k
`scale=-1:720` : resize to 720p

`scale={width}:{height}`, `-1` means automatic and keeps the aspect ratio. Note that the size must be a multiple of 2 otherwise an error will occur, see https://trac.ffmpeg.org/wiki/Scaling

**Reply by kaelsonofkrypto** [reply]
so then if i use a 4k source that is 3840x2160 or 3840x1634 (cropped), and i manually enter 3840 in the stereo width processing field, then this would result in a wrong option and impact the 3d negatively? or are you saying that in the past specifying higher than 1920 would have been wrong but now that u have added that stereo field i can enter 3840 and it would work. 

i assumed u entered the stereo width field to enter based on the total width, so for example im doing a 4k source half sbs, since its technically 1920 per eye the total width would be 3840. so would that 3840 be what i would enter in the stereo width field? now lets say im doing a 4k full sbs, the total output would be 3840 per eye which would be a total width of 7680, so in this instance, would 7680 be what i enter in stereo width for a 4k full sbs? Or am i confused and is this stereo width actually based on per eye so does 1920 dropdown refer to 1920 per eye which would be what i am doing 4k half sbs? and then lets say if i do 4k full sbs then should i be entering 3840 in that stereo option field?  Also i had assumed that stereo width deals with the horizontal offset between the left and right eye images but somehow u setup in iw3 to simplify it for user and it automatically figures that out based off the width of the file u specified in that section.

Im mainly using quest 3 to watch 3d on virtual desktop. and so also i want to confirm the output limit would that not apply in the situation im doing since it will just output based on the source im using if i dont put anything in that field and leave it blank?

**Reply by nagadomi** [reply]
It is wrong to enter a value higher than 1920 for `Stereo Processing Width`.
Usually `Default`(no resize) is the right choice. 
`Stereo Processing Width` is a feature used to downscale the depth image when the width of the depth image is greater than 1920 for yml import. https://github.com/nagadomi/nunif/discussions/169

**Reply by kaelsonofkrypto** [reply]
INDEPTH BACKGROUND
sorry for not being more clear and specifying the full scope, hopefully below this gives a more indepth reason for u to understand why i am asking these questions and what im actually asking>

 (DURING EXPORT DISPARITY PHASE)
 what i did was a clever work around because i know u cant do a depth resolution of 2160 during export disparity, i would tyically use default in stereo processing and 518 depth resolution, because it will degrade the depth map above 518. 
 
 after iw3 would render the maps, i would then import the maps in davinci, deflicker them, improve gamma and dehaze to fine tune the iw3 maps and upscale the maps to 3840x2160, which made the edges better.
 
 (DURING IMPORT OF YML PHASE)
 Afterwards i would delete the maps in the iw3 depth folder and replace them with my upscaled maps and during input of yml to convert to sbs i would enter 2160 in the depth resolution field and 3840 in stereo width 
 
 QUESTIONS
 i didnt know that stereo processing field was a downscale field, 
  
 My reasons for asking these questions is because i do not want to affect the upscaled maps and quality of the resolution im using which was the reason i upscaled the depth maps outside in davinci since i already new that was the limitation of the model.
 
 so just for clarification to make sure i understand you correctly, based on your previous response, stereo processing is not used during export disparity, its only used for yml input to sbs, and if i select default, this will not downscale my 4k upscaled depth maps?

**Reply by nagadomi** [reply]
`row_flow_v3` may not correctly process depth images that have a width greater than 1920.  `Stereo Processing Width` is a setting to work around this problem.

About `Stereo Processing Width`,
when `Default` is specified, the depth images will not be resized (do nothing).
If any numerical value is specified, the depth images will be resized to the specified size before being input into `row_flow_v3` model. If the depth image is larger than the specified size, it will be downscaled; if it is smaller, it will be upscaled.

`Stereo Processing Width` is used when normal SBS conversion and when Import. It is not used when Export.
`Depth Resolution` is used when normal SBS conversion and when Export. It is not used when Import.

**Reply by kaelsonofkrypto** [reply]
Thanks for your quick response, i feel so bad for asking so many questions lol but i have to know,  have one last question about this topic, is the 1920 width u refer to based on total width or PER EYE because this is an important distinction, if your 1920 is per eye, then this means this is 4k half sbs or 1080p full sbs that can be done on total output, but if your 1920 means total width then this mean its actually 720p full sbs or 1080p half sbs is the total output we can work with at max. Thats something important i want to also get clarification on?

Understanding which scenario applies will help make informed decisions about the resolutions i and others can work with when determining max sources using the iW3 tool, ensuring the best possible results for 3D conversion.

**Reply by nagadomi** [reply]
1920 limitation is about the size(width) of the **depth image**. It is PER EYE of those two options.
However, it does not limit the output size. 4K and 8K frame(rgb) image is acceptable.
This limitation is only about the size of the **depth image**.

**Comment by kaelsonofkrypto** [comment]
i think i got what u are saying
To sum it up, workflow can accommodate the following:
Upscaled depth map can be 4K resolution (3840x2160) if i am going to do a half sbs conversion or i can do the 1920p depth map if i am going to do full sbs 
far as RGB source it can be any resolution up to 8K resolution
and selecting default on stereo depth processing will ensure the depth map isn't downscaled and the best possible quality is maintained during the 3D conversion process.

**Reply by nagadomi** [reply]
If the depth image is 4k or 8k, it can be downscaled on iw3 side using `Stereo Processing Width` option.
In this case, you can try various depth sizes (Default or 4k, 2k, 1920 or 1280).

**Reply by kaelsonofkrypto** [reply]
thanks for answering all my questions, and taking the time out to respond,  i do appreciate that.
